src/shared/tpm2-util.c

```diff
@@ -311,55 +321,222 @@ static int tpm2_credit_random(ESYS_CONTEXT *c) {
         return 0;
 }
 
-static int tpm2_make_primary(
-                ESYS_CONTEXT *c,
-                ESYS_TR *ret_primary,
-                TPMI_ALG_PUBLIC alg,
-                TPMI_ALG_PUBLIC *ret_alg) {
+const TPM2B_PUBLIC *tpm2_get_primary_template(bool use_srk_model, bool is_ecc) {
```

Comment:
maybe make this a flags param?

```c
typedef enum Tpm2SRKTemplateFlags = {
        TPM2_SRK_TEMPLATE_NEW_STYLE = 1 << 0,
        TPM2_SRK_TEMPLATE_ECC       = 1 << 1,
        _TPM2_SRC_TEMPLATE_MAX      = TPM2_SRK_TEMPLATE_NEW_STYLE|TPM2_SRK_TEMPLATE_ECC,
} Tpm2SRKTemplateFlags;
```

And then:

```c
const TPM2B_PUBLIC *tpm2_get_primary_template(Tpm2SRKTemplateFlags flags) {
        static const TPM2B_PUBLIC templ[_TPM2_SRC_TEMPLATE_MAX] = {
                …
        };
        assert(flags >= 0);
        assert(flags < _TPM2_SRC_TEMPLATE_MAX);
        return templates[flags];  
 }
```

this of course is only efficient if the template space is dense, i.e. for each flags combo we have a distinct template. But so far we have...

using a flags concept makes calling this function a lot more readable as flags names are not as opaque as just "true" and "false"